mismi-s3/main/s3.hs

```diff
@@ -127,48 +85,20 @@ runC c = do
       getSize a >>= liftIO . maybe exitFailure (putStrLn . show)
     Sync s d m f ->
       syncWithMode m s d f
+    List a rq ->
+      rec (list a) (listRecursively a) rq >>= liftIO . mapM_ (putStrLn . unpack . addressToText)
```

Comment:
Can this be the same as the previous `listRecursively'`implementation, using `list'` and `listRecursively'`. Streaming is very useful :dancers: